build/tasks/test-on-node.js

```diff
@@ -57,39 +88,63 @@ module.exports = function( grunt ) {
 	function registerEvents( QUnit, file, runEnd ) {
 		var runDone = false;
 		var testActive = false;
+		var stats = {
+			passed: 0,
+			failed: 0,
+			skipped: 0,
+			todo: 0
+		};
 
 		QUnit.begin( function() {
 			grunt.log.ok( "Testing " + file + " ..." );
 		} );
+
 		QUnit.testStart( function() {
 			testActive = true;
 		} );
+
 		QUnit.log( function( details ) {
-			if ( !testActive || details.result ) {
+			if ( !testActive || details.result || details.todo ) {
 				return;
 			}
 			var message = "name: " + details.name + " module: " + details.module +
 				" message: " + details.message;
 			grunt.log.error( message );
 		} );
-		QUnit.testDone( function() {
+
+		QUnit.testDone( function( details ) {
 			testActive = false;
+
+			if ( details.skipped ) {
+				stats.skipped++;
+			} else if ( details.failed && !details.todo ) {
```

Comment:
Todo tests can also "fail". I think this would be clearest if it mirrored the HTML reporter:
```js
// This test passed if it has no unexpected failed assertions
var testPassed = details.failed > 0 ? details.todo : !details.todo;

if ( details.skipped ) {
	stats.skipped++;
} else if ( !testPassed ) {
	stats.failed++;
} else if ( details.todo ) {
	stats.todo++;
} else {
	stats.passed++;
}
```